plugins/catalog-backend/src/service/CatalogBuilder.ts

```diff
@@ -430,6 +431,13 @@ export class CatalogBuilder {
     return this;
   }
 
+  withProcessingEngine(
+    processingEngine: CatalogProcessingEngine,
+  ): CatalogBuilder {
+    this.processingEngine = processingEngine;
```

Comment:
I'm curious what the motivation is behind this. The `CatalogProcessingEngine` type only has `start` and `stop`, intentionally stripped down to its absolute minimum. It doesn't even accept the passed-in orchestrator, stitcher, etc etc so its value standalone is probably very small right?

As a matter of fact, we were on a trajectory to [completely detaching](https://github.com/backstage/backstage/pull/18062/files#diff-98a36cbcbd563311ebfa4f964c3de8cbaf2e235520d57a95ce88d77b06d59eceR48) the implementation of the processing engine from [what gets returned](https://github.com/backstage/backstage/pull/18062/files#diff-3320c1fccf36da64786791589cd4f2099717b687b4b45483a35a61e328fd7fbcR580). It is instead the higher level concept of "background processes that sometimes should run, sometimes not".

Maybe we can find a different place to achieve what you wanted to target?

Reply:
Maybe I'm misunderstanding, but I thought the processingEngine is responsible for determining if an entity should be processed or not.

I am looking to replace the default behavior with one that based on a predicate would replace a conflicting entity with a new one rather than reject the new one.

Reply:
Ah, is it something in this direction? https://github.com/backstage/backstage/pull/16660

Reply:
And the `DefaultCatalogProcessingEngine` does a lot of things. This would allow me to copy it and make some modifications and easily inject it. But again, this may be the wrong level to modify it on.

Reply:
Yeah I think it's prolly better to try to solve for the specific use-case is possible first. We used to have some parts of the catalog internals be external interfaces that could be replaced, but it makes it very hard to evolve things.

What conflict resolution logic do you have in mind? Is it even something that would help to have as a default?